ui/hooks/src/use-close-notification.ts

```diff
@@ -0,0 +1,34 @@
+import { useState } from 'react';
+
+export function useCloseNotification(key: string, initialValue = false) {
+  const [closed, setClosed] = useState(() => {
+    if (window?.localStorage) {
+      try {
+        const closedFlag = window.localStorage.getItem(key);
+        if (closedFlag) {
+          return JSON.parse(closedFlag);
+        } else {
+          window.localStorage.setItem(key, JSON.stringify(initialValue));
+          return initialValue;
+        }
+      } catch (err) {
+        // console.log(err);
+        return initialValue;
+      }
+    }
+  });
+
+  const setValue = value => {
+    try {
+      // Allow value to be a function so we have same API as useState
+      const newClosedValue = value instanceof Function ? value(closed) : value;
```

Comment:
If I understand correctly you want to support callback based state setter here.
The correct approach would be to use the actual previous state from the setter function:
```
setClosed(prev => {
  const result = typeof value === 'function' ? value(prev): value;
  // set the localstorage..
  return result 
});
```
But i don't feel like we need this functionality... I would prefer to have simple logic instead, so keep it simple, don't replicate the apis, provide good typings and that's it :)